ethcore/light/src/transaction_queue.rs

```diff
@@ -202,14 +223,20 @@ impl TransactionQueue {
 	/// Get all transactions ready to be propagated.
 	/// `best_block_number` and `best_block_timestamp` are used to filter out conditionally
 	/// propagated transactions.
+	///
+	/// Returned transactions are batched by sender, in order of ascending nonce.
 	pub fn ready_transactions(&self, best_block_number: u64, best_block_timestamp: u64) -> Vec<PendingTransaction> {
-		self.by_account.values().flat_map(|acct_txs| {
-			acct_txs.current.iter().take_while(|tx| match tx.condition {
-				None => true,
-				Some(Condition::Number(blk_num)) => blk_num <= best_block_number,
-				Some(Condition::Timestamp(time)) => time <= best_block_timestamp,
-			}).cloned()
-		}).collect()
+		self.by_account.values()
+			.flat_map(|acct_txs| {
+				acct_txs.current.iter().take_while(|tx| match tx.condition {
+					None => true,
+					Some(Condition::Number(blk_num)) => blk_num <= best_block_number,
+					Some(Condition::Timestamp(time)) => time <= best_block_timestamp,
+				}).map(|info| info.hash)
+			})
+			.filter_map(|hash| self.by_hash.get(&hash))
```

Comment:
I chose to use `filter_map` specifically to avoid panics in this case. I think it's better for something not to be propagated than for the whole node to go down. Adding a warning here "Inconsistency detected between `by_account` and `by_hash`" makes sense to me, though.